mmcls/models/necks/dolg.py

```diff
@@ -0,0 +1,219 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+from typing import Optional, Sequence
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from mmengine.model import BaseModule
+from mmengine.utils import is_seq_of
+
+from mmcls.registry import MODELS
+from .gem import GeneralizedMeanPooling
+
+
+class MultiAtrous(BaseModule):
+    """Multiple atrous convolution layers.
+
+    Args:
+        in_channel (int): Number of channels in the input.
+        out_channel (int): Number of channels in the output.
+        dilation_rates (Sequence[int]): The list of the dilation rates
+            of multiple atrous convolution. Defaults to `[3, 6, 9]`.
+        init_cfg (dict, optional): dictionary to initialize weights.
+            Defaults to None.
+    """
+
+    def __init__(self,
+                 in_channel: int,
+                 out_channel: int,
+                 dilation_rates: Sequence[int] = [3, 6, 9],
+                 init_cfg: Optional[dict] = None):
+        super(MultiAtrous, self).__init__(init_cfg)
+        assert is_seq_of(
+            dilation_rates,
+            int), ('``dilation_rates`` must be a sequence of int.')
+        assert out_channel % 2 == 0
+
+        # In the original implementation, here hidden-dim was set to a
+        # fixed 512 as it was designed for resnet101. We set it here to
+        # half of out_c for the sake of generality, which is consistent
+        #  when using resnet101.
+        hidden_channel = out_channel // 2
+        self.dilated_conv_list = nn.ModuleList([
```

Comment:
https://github.com/feymanpriv/DOLG/blob/63b117d76f660db9617c2081ec771e3cbdf4bcc8/core/model/dolg_model.py#L143
`dilation=1` conv may be needed.

Reply:
yes, and the kernel_size should be 1 too.

Reply:
https://github.com/feymanpriv/DOLG/blob/63b117d76f660db9617c2081ec771e3cbdf4bcc8/core/model/dolg_model.py#L143 is for `self.gap_conv`, its dilation is 1.

self.dilated_conv_lists' `dilation`  are `rate` refer to https://github.com/feymanpriv/DOLG/blob/63b117d76f660db9617c2081ec771e3cbdf4bcc8/core/model/dolg_model.py#L147.

To avoid confusion in reading the code, I will put them in reverse order